c7n/resources/mq.py

```diff
@@ -82,6 +83,57 @@ def get_dimensions(self, resource):
         return [{'Name': self.model.dimension,
                  'Value': "{}-1".format(resource['BrokerName'])}]
 
+@MessageBroker.filter_registry.register('vpc')
+class VpcFilter(VpcFilter):
+
+    sg_vpc_dict = {}
+
+    def get_related_ids(self, resources):
+        client = local_session(self.manager.session_factory).client('ec2')
+        related_ids = []
+        for r in resources:
+            sg_ids = r.get('SecurityGroups', [])
+            vpc_id = self.sg_vpc_dict.get(sg_ids[0])
+            if(vpc_id):
+                related_ids.append(vpc_id)
+                continue
+            try:
+                response = client.describe_security_groups(
+                    GroupIds=[sg_ids[0]],
+                )
+                vpc_id = response.get('SecurityGroups')[0].get('VpcId')
+                related_ids.append(vpc_id)
+                self.sg_vpc_dict[sg_ids[0]] = vpc_id
+            except ClientError: 
+                continue 
+        return set(related_ids)
```

Comment:
If this is going to be a set, then line 93 can use `set()` to create an empty set and line 98 can use `.add()` to update the set. Then, no conversion from list to set is required.

Reply:
Sounds good, I'll make this change.